### DISCUSSION TITLE
Purpose of DuckDB PySpark API and Perhaps a More Useful Alternative

### DISCUSSION THREAD

**Discussion by erikhuck** [duckdb/duckdb#9679] [Q&A] [answered]
From what I can tell, the DuckDB PySpark API is a re-implementation of the PySpark API. I'm not sure why this was decided. And I see that you have to have a dataframe or data ready already in order to use it, which seems to defeat the point of using PySpark. There doesn't appear to be a way to read from a DuckDB database, which seems to be the whole point of using duckdb. I am pretty confused about the intention behind these experimental features.

Anyway, PySpark enables custom connections via JDBC, which would allow connecting to a DuckDB database. However, it'd be helpful if the duckdb package could streamline that by providing a PySpark session (an actual PySpark session, not a re-implementation) given only a duckdb database file or duckdb connection. Then one would be able to use PySpark with the data in the duckdb database and re-use the actual PySpark API rather than re-implementing everything. This would be especially useful if one didn't have to download the duckdb JDBC.

**Comment by Tishj** [comment, marked as answer]
The purpose of the Spark API is to offer an alternative backend using existing spark code.

The idea is not to create interop between spark and duckdb, cool that you found a way to do so using jdbc though.

We are not planning to break existing spark behavior.
